sdk/storage/Azure.Storage.DataMovement.Blobs/README.md

```diff
@@ -40,7 +40,178 @@ az storage account create --name MyStorageAccount --resource-group MyResourceGro
+```C# Snippet:UploadSingle_ConnectionStringAsync
+    // Create simple transfer single blob upload job
+    DataTransfer dataTransfer = await transferManager.StartTransferAsync(
+        sourceResource: new LocalFileStorageResource(sourceLocalPath),
+        destinationResource: new BlockBlobStorageResource(destinationBlob));
+    await dataTransfer.AwaitCompletion();
+}
+```
+
+```C# Snippet:DownloadSingle_SharedKeyAuthAsync
+    // Simple Download Single Blob Job
+    StorageResource sourceResource = new BlockBlobStorageResource(sourceBlob);
+    StorageResource destinationResource = new LocalFileStorageResource(downloadPath);
+    await transferManager.StartTransferAsync(
+        sourceResource,
+        destinationResource);
+
+    StorageResource sourceResource2 = new BlockBlobStorageResource(sourceBlob);
+    StorageResource destinationResource2 = new LocalFileStorageResource(downloadPath2);
+
+    await transferManager.StartTransferAsync(
+        sourceResource: new BlockBlobStorageResource(sourceBlob, new BlockBlobStorageResourceOptions()
+            {
+                DownloadOptions = new BlobStorageResourceDownloadOptions()
+                {
+                    Conditions = new BlobRequestConditions(){ LeaseId = "xyz" }
+                }
+            }),
+        destinationResource: new LocalFileStorageResource(downloadPath2));
+}
+```
+
+```C# Snippet:UploadDirectory_SasAsync
+    // Create simple transfer directory upload job which uploads the directory and the contents of that directory
+    DataTransfer dataTransfer = await transferManager.StartTransferAsync(
+        sourceResource: new LocalDirectoryStorageResourceContainer(sourcePath),
+        destinationResource: new BlobDirectoryStorageResourceContainer(container, Randomize("sample-blob-directory")),
+        transferOptions: options);
+}
+```
+
+```C# Snippet:UploadDirectory_CompletedEventHandler
+    // Create BlobTransferManager with event handler in Options bag
+    TransferManagerOptions options = new TransferManagerOptions();
+    ContainerTransferOptions containerTransferOptions = new ContainerTransferOptions();
+    containerTransferOptions.SingleTransferCompleted += (SingleTransferCompletedEventArgs args) =>
+    {
+        // Customers like logging their own exceptions in their production environments.
+        using (StreamWriter logStream = File.AppendText(logFile))
+        {
+            logStream.WriteLine($"File Completed Transfer: {args.SourceResource.Path}");
+        }
+        return Task.CompletedTask;
+    };
+    TransferManager transferManager = new TransferManager(options);
+
+    // Create simple transfer directory upload job which uploads the directory and the contents of that directory
+    DataTransfer uploadDirectoryJobId = await transferManager.StartTransferAsync(
+        new LocalDirectoryStorageResourceContainer(sourcePath),
+        new BlobDirectoryStorageResourceContainer(container, Randomize("sample-blob-directory")));
```

Comment:
We can take this part out and make it, it's own snippet so we can show how to do a directory transfer.

Reply:
Also lets remove the `Randomize` part too again